km61/Morozov_Artem/laboratory1.sql

```diff
@@ -1,2 +1,55 @@
--- LABORATORY WORK 1
--- BY Morozov_Artem
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+створювати таблиці
+4 бали
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE user student
+IDENTIFIED by Artem;
+GRANT create table to  Artem;
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент має мобільний номер іноземного оператора.
+4 бали
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE TABLE  student_name (
+  s_name VARCHAR (30), not null;
+  telephon_number integer (10), not null;
+);
```

Comment:
azaza). Cool. 
First question to your entities: what if student has more then 1 telephone number?) But it's still very far from task).
Okay, we have a relationships: many to many, because every user can has more than one telephone number of foreign operator and foreign operators have more than one student that has their telephone number.
You have to create next entities: student, telephone_number, operator, operator_has_id and student_has_telephone_number.
What are you need to do:
1. Student entity must contain of student_id (PK) and other information about student which you need. Relax, it doesn't matter).
2. Operator Entity have to consist of operator_name as PK, contry_name_of_operator and that's all. Because operator can has more then one id's (as example: 063, 093)
3. Operator_has_id entity must consist of operator_id as PK and operator _name as Foreign Key which reference to PK of previous Entity. It's also
4. Telephone_number entity must contain of next attributes: phone_number PK ( yeaah, yeeah, it's Primary Key) and operator_id. Operator_id it's FK that references Operator_has_id entity.
5. Student_has_telephone_number must consist of phone_number as PK and student_id as FK. And yes!, elements of this entity establish a correspondence between student and country of operator.
And now question to teacher: What a sence of this task?))) It's not understandable. Abstraction it's great, but what we need to do in this task? I assume , that we need to create a entities which bind one entity to other , am i right? But for what we have a condition here like: (Студент має мобільний номер іноземного оператора)? What we had to do? Create a entity: student_has_a_foreign_operator_phone_number?)))00